documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
+.4+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel7l
-make bcm2711_defconfig
+$ cd linux
+$ KERNEL=kernel7l
+$ make bcm2711_defconfig
 ----
+| Raspberry Pi 400
+| Raspberry Pi Compute Module 4S
+| Raspberry Pi Compute Module 4
+
 
-For Raspberry Pi 3, 3+, 4, 400 and Zero 2 W, and Raspberry Pi Compute Modules 3, 3+ and 4 default 64-bit build configuration:
-[,bash]
+.10+^.^| 64-bit
+| Raspberry Pi 3
+.9+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel8
-make bcm2711_defconfig
+$ cd linux
+$ KERNEL=kernel8
+$ make bcm2711_defconfig
 ----
+| Raspberry Pi 3+
+| Raspberry Pi 4
+| Raspberry Pi 400
+| Raspberry Pi Zero 2 W
+| Raspberry Pi Compute Module 3
+| Raspberry Pi Compute Module 3+
+| Raspberry Pi Compute Module 4S
+| Raspberry Pi Compute Module 4
 
-For Raspberry Pi 5 default 64-bit build configuration:
-[,bash]
+
+| Raspberry Pi 5
+.1+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel_2712
-make bcm2712_defconfig
+$ cd linux
+$ KERNEL=kernel_2712
+$ make bcm2712_defconfig
 ----
+|===
 
-===== Customising the kernel version using `LOCALVERSION`
+===== Customise the kernel version using `LOCALVERSION`
 
-In addition to your kernel configuration changes, you may wish to adjust `LOCALVERSION` to ensure your new kernel does not receive the same version string as the upstream kernel. This clarifies that you are running your own kernel in the output of `uname`, and ensures that existing modules in `/lib/modules` are not overwritten.
+Adjust `LOCALVERSION` to clarify that you are running your own kernel in the output of `uname` and prevent the kernel from overwriting existing modules in `/lib/modules`.
 
 To adjust `LOCALVERSION`, change the following line in `.config`:
 
+[source,ini]
 ----
 CONFIG_LOCALVERSION="-v7l-MY_CUSTOM_KERNEL"
 ----
 
-You can also change this setting graphically as shown in xref:linux_kernel.adoc#configuring-the-kernel[the kernel configuration instructions]. It is located in "General setup" \=> "Local version - append to kernel release".
+You can also change this setting graphically with `menuconfig` at *General setup* > *Local version - append to kernel release*. For more information about `menuconfig`, see xref:linux_kernel.adoc#configure-the-kernel[the kernel configuration instructions].
+
+==== Build and install the kernel natively
+
+Next, build and install the kernel, modules, and Device Tree blobs into the boot partition of your Raspberry Pi. This step can take a long time depending on the Raspberry Pi model you are using.
 
-==== Building the kernel
+TIP: If you don't want to install the freshly-compiled kernel onto the Raspberry Pi where you run this command, copy the compiled kernel to the boot partition of a separate boot media instead of `/boot/firmware/`.
 
-Build and install the kernel, modules, and Device Tree blobs. This step can take a long time depending on the Raspberry Pi model you are using.  For the 32-bit kernel:
+To build and install the 64-bit kernel, run the following commands:
```

Comment:
The "configure" section described 32-bit before 64-bit, so feels a bit weird that the "build' section describes 64-bit before 32-bit? :confused: